mlflow/catboost.py

```diff
@@ -134,6 +134,11 @@ def save_model(
     with open(os.path.join(path, conda_env_subpath), "w") as f:
         yaml.safe_dump(conda_env, stream=f, default_flow_style=False)
 
+    pip_req_subpath = "requirements.txt"
+    pip_deps = _get_additional_pip_dep(conda_env)
+    with open(os.path.join(path, pip_req_subpath), "w") as f:
```

Comment:
Since the code is exactly the same in each flavor, would it make sense to factor out the whole code snippet into `mlflow.utils.environment`